pkg/server/dns_server.go

```diff
@@ -142,39 +152,76 @@ func (h *DNSServer) handleACMETXTChallenge(zone string, m *dns.Msg) error {
 func (h *DNSServer) handleACNAMEANY(zone string, m *dns.Msg) {
 	nsHeader := dns.RR_Header{Name: zone, Rrtype: dns.TypeNS, Class: dns.ClassINET, Ttl: h.timeToLive}
 
-	resultFunction := func(ipAddress net.IP) {
+	resultFunction := func(zone string, ipAddress net.IP) {
 		m.Answer = append(m.Answer, &dns.A{Hdr: dns.RR_Header{Name: zone, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: h.timeToLive}, A: ipAddress})
+		dotDomains := []string{zone, dns.Fqdn(h.options.Domains[0])}
```

Comment:
Why are we using zone and the first domain specified by user during result creation? Ideally, won't we only want to use the domain received in the interaction?